test/unit/range/view/view_interleave_test.cpp

```diff
@@ -0,0 +1,58 @@
+#include <seqan3/io/stream/debug_stream.hpp>
+#include <seqan3/range/view/all.hpp>
+#include <seqan3/range/view/interleave.hpp>
+#include <seqan3/range/view/take.hpp>
+#include <seqan3/std/view/reverse.hpp>
+
+#include <gtest/gtest.h>
+
+using namespace seqan3;
+
+TEST(view_interleave, basic)
+{
+    std::string u{"FOOBARBAXBAT"};
+    std::string i{"in"};
+    size_t s = 3;
+    std::string cmp{"FOOinBARinBAXinBATin"};
+    std::string cmp_rev{"niTAB"};
+
+    // pipe notation
+    // explicitly call view::all
+    auto v0 = view::all(u) | view::interleave(s, view::all(i));
+    EXPECT_TRUE(ranges::equal(cmp, v0));
+    // don't call view::all
+    auto v1 = u | view::interleave(s, i);
+    EXPECT_TRUE(ranges::equal(cmp, v1));
+
+    // function notation
+    // explicitly call view::all
+    auto v2{view::interleave(view::all(u), s, view::all(i))};
+    EXPECT_TRUE(ranges::equal(cmp, v2));
+    // don't call view::all
+    auto v3{view::interleave(u, s, i)};
+    EXPECT_TRUE(ranges::equal(cmp, v3));
+
+    //combinability
+    // explicitly call view::all
+    auto v4 = view::all(u) | view::interleave(s, view::all(i)) | view::reverse | view::take(5);
+    EXPECT_TRUE(ranges::equal(cmp_rev, v4));
+    // don't call view::all
+    auto v5 = u | view::interleave(s, i) | view::reverse | view::take(5);
+    EXPECT_TRUE(ranges::equal(cmp_rev, v5));
+}
+
+TEST(view_interleave, concepts)
+{
+    std::string u{"FOOBARBAXBAT"};
+    std::string i{"in"};
+    size_t s = 3;
+    auto v1 = detail::view_interleave(view::all(u), s, view::all(i));
+
+    EXPECT_TRUE(std::ranges::InputRange<decltype(v1)>);
+    EXPECT_TRUE(std::ranges::ForwardRange<decltype(v1)>);
+    EXPECT_TRUE(std::ranges::BidirectionalRange<decltype(v1)>);
+    EXPECT_TRUE(std::ranges::RandomAccessRange<decltype(v1)>);
+    EXPECT_TRUE(std::ranges::View<decltype(v1)>);
+    EXPECT_TRUE(std::ranges::SizedRange<decltype(v1)>);
+    EXPECT_TRUE(std::ranges::CommonRange<decltype(v1)>);
+}
```

Comment:
Please remove all the `view::all` in here, that should be taken care of automatically. Also pick a second example as original input that does not mode RandomAccess or Sized so that you can check that the other path (that forwards to view::split...) is actually taken. Also verify the properties on that view as I documented them in the table above!

Reply:
Would you happen to have an example of something which doesn't model RandomAccess or Sized?

Reply:
std::list doesn't model randomaccess, view::filter models neither randomaccess nie sized!

Reply:
So, testing this, I now have a question. We implemented it using view::split. I think we should have used view::chunk instead? I think view::split splits matching a pattern, not at a specified interval.

Reply:
> We implemented it using view::split. I think we should have used view::chunk instead? I think view::split splits matching a pattern, not at a specified interval.

True, and since there is no test testing this, we also didnt realise now.

Reply:
Please also test explicitly that the other concepts are false :nail_care:

Reply:
:+1: